test/e2e/operator/garden/create_rotate_delete.go

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) 2022 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package garden
+
+import (
+	"context"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	"github.com/gardener/gardener/test/e2e/operator/garden/internal/rotation"
+	rotationutils "github.com/gardener/gardener/test/utils/rotation"
+)
+
+var _ = Describe("Garden Tests", Label("Garden", "default"), func() {
+	var (
+		backupSecret = defaultBackupSecret()
+		garden       = defaultGarden(backupSecret)
+	)
+
+	It("Create Garden, Rotate Credentials and Delete Garden", Label("credentials-rotation"), func() {
+		By("Create Garden")
+		ctx, cancel := context.WithTimeout(parentCtx, 3*time.Minute)
+		defer cancel()
+
+		Expect(runtimeClient.Create(ctx, backupSecret)).To(Succeed())
+		Expect(runtimeClient.Create(ctx, garden)).To(Succeed())
```

Comment:
We also don't do this in the other e2e tests since it also can have the advantage that one might be able to debug what was going wrong. When the resources get auto-deleted, this is not possible or significantly harder.